### PR TITLE
Python: Adding core HttpSkill

### PR DESCRIPTION
**Pull request by JTremb**
### Motivation and Context
HttpSkill  python version

### Description

- Porting the HttpSkill from C# to Python.
- Adding corresponding unit tests.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/core_skills/http_skill.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        returns:
+            The response body as a string.
+        """
+        has_body, body = context.variables.get("body")
+        if not url:
+            raise ValueError("url cannot be `None` or empty")
+        if not has_body:
+            raise ValueError("no Body found in context")
+        if not body:
+            raise ValueError("Body of the POST request cannot be None")
```
Comment (mkarle) [inline python/semantic_kernel/core_skills/http_skill.py:89]: This is a PUT

File: python/semantic_kernel/core_skills/http_skill.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        returns:
+            The response body as a string.
+        """
+
+        has_body, body = context.variables.get("body")
+        if not url:
+            raise ValueError("url cannot be `None` or empty")
+        if not has_body:
+            raise ValueError("no Body found in context")
+        if not body:
```
Comment (mkarle) [inline python/semantic_kernel/core_skills/http_skill.py:61]: A body isn't required for POSTs or PUTs.

**Review by mkarle — CHANGES_REQUESTED** [review]

**Review by ankit-gautam23 — COMMENTED** [review]
Raising Value Error for body of puts and posts are not needed. Everything else looks fine.

File: python/semantic_kernel/core_skills/http_skill.py (hunk 2 above)
Comment (JTremb) [reply python/semantic_kernel/core_skills/http_skill.py]: thanks updated!

File: python/semantic_kernel/core_skills/http_skill.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    async def put_async(self, url: str, context: SKContext) -> str:
+        """
+        Sends an HTTP PUT request to the specified URI and returns
+        the response body as a string.
+        params:
+            url: The URI to send the request to.
+        returns:
+            The response body as a string.
+        """
+        _, body = context.variables.get("body")
```
Comment (awharrison-28) [inline python/semantic_kernel/core_skills/http_skill.py:79]: same nit as above

**Review by awharrison-28 — COMMENTED** [review]
@JTremb can you also update the Core Skills Feature Matrix https://github.com/microsoft/semantic-kernel/blob/main/FEATURE_MATRIX.md#core-skills

Other than that and a couple of non-blocking nits, LGTM

File: python/semantic_kernel/core_skills/http_skill.py (hunk 2 above)
Comment (awharrison-28) [reply python/semantic_kernel/core_skills/http_skill.py]: Did you push the changes? I'm still seeing errors raised if a body is not found for the post request

**Comment by JTremb** [comment]
@awharrison-28  Thanks! updated.

File: python/semantic_kernel/core_skills/http_skill.py (hunk 2 above)
Comment (JTremb) [reply python/semantic_kernel/core_skills/http_skill.py]: that is indeed likely that I forgot to push the changes, should be fine now.

**Review by awharrison-28 — APPROVED** [review]

**Comment by MovGP0** [comment]
Please fix the double empty lines and whitespace changes that don't contribute to the functionality of the code.

Consider a dedicated PR for whitespace changes that improve code readability.

**Pull request merged**